web-common/src/features/entity-management/actions.ts

```diff
@@ -35,24 +34,22 @@ export async function renameFileArtifact(
 
 export async function deleteFileArtifact(
   instanceId: string,
-  name: string,
+  filePath: string,
   type: EntityType,
-  names: Array<string>,
+  allPaths: Array<string>,
   showNotification = true,
 ) {
-  const path = getFileAPIPathFromNameAndType(name, type);
+  const name = extractFileName(filePath);
   try {
-    await runtimeServiceDeleteFile(instanceId, removeLeadingSlash(path));
+    await runtimeServiceDeleteFile(instanceId, removeLeadingSlash(filePath));
 
     httpRequestQueue.removeByName(name);
     if (showNotification) {
       notifications.send({ message: `Deleted ${getLabel(type)} ${name}` });
     }
 
     if (get(appScreen)?.name === name) {
-      const route = getRouteFromName(getNextEntityName(names, name), type);
-
-      await goto(route);
+      await goto(getNextEntityName(allPaths, name));
```

Comment:
When we migrate, navigation to the "next entity name" will be confusing in cases where users do not have a `sources/`, `models/`, `dashboards/` directory structure. So instead, to avoid too much magic in the code, I think we should navigate to the root `/` in all cases.

Reply:
Hmm true. I thought we could navigate to the next file in the folder. But i havent seen any editor do that.